### PR TITLE
docs: add content for module 3

### LINKED ISSUE #43: Proposal to archive this repository
As part of the 2026 community goals (more details here: https://github.com/asyncapi/community/issues/3495), we conducted an audit of all repositories. The summary of the findings can be found here: https://github.com/asyncapi/community/issues/3495#issuecomment-4681625773.

As suggested by @deberg in the following comment:
https://github.com/asyncapi/community/issues/3495#issuecomment-4722354038

we are proposing to archive this repository.

I am opening this issue to gather feedback and approval from the code owners of this repository.

Tagging all code owners for their feedback and approval.
@alequetzalli @quetzalliwrites

### PR DESCRIPTION
**Pull request by mhmohona**
**Description**
Added content for module 3 -

> AsyncAPI Specification  (SCENE) Eve helps Chan program his spaceship, so they can stay in touch.
Deep dive into various components of AsyncAPI: info, channels, components, tags, etc.
How to define operations (like send and address), messages, and schemas.

**Related issue(s)**
https://github.com/asyncapi/website/issues/1520

### CODE REVIEW

File: 100-level-AsyncAPI-Intro/step2.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
+## Deep Dive into Various Components of AsyncAPI
+
+In the control room of Chan's spaceship, Eve is assisting Chan in setting up the communication system using AsyncAPI. They're working together to ensure they can maintain contact even when Chan is light-years away. Eve explains the different components of AsyncAPI as they program each part.
+
+## Understanding AsyncAPI Components through Chan and Eve's Programming Session
+
```
Comment (TRohit20) [inline 100-level-AsyncAPI-Intro/step2.md]: I think you can remove these lines as I begin the module with a similar storyline, touch lightly on structure and lead to `deep dive of various components`, the rest fits in well and needs to modification for now as I see.

You can ref #9 to get a better sense of the suggestion made

**Comment by quetzalliwrites** [comment]
Hey folks, so please update your planet names to:
- Eve's planet: **Capuccinova**
- Chan's planet: **Brownieterra**

**Review by derberg — COMMENTED** [review]
lgtm

only remark is the host of the server: `space.mosquitto.org`

it is not real, and also afaik in killercoda it is not possible to go outside the killercoda network, this is why in my interactive tutorial I started MQTT broker locally in killercoda container, so I think you need to use localhost - cause you will not be able to use sample document for code generation or other things in killercode

on to you @alequetzalli

**Comment by quetzalliwrites** [comment]
@mhmohona ping ping, please make sure to address the important feedback that @derberg left last week :)

File: 100-level-AsyncAPI-Intro/step2.md [hunk 2]
```diff
@@ -0,0 +1,244 @@
...
+## Understanding AsyncAPI Components through Chan and Eve's Programming Session
```
Comment (quetzalliwrites) [inline 100-level-AsyncAPI-Intro/step2.md]:
```suggestion
## Understanding AsyncAPI components 
```

File: 100-level-AsyncAPI-Intro/step2.md [hunk 3]
```diff
@@ -0,0 +1,244 @@
...
+
+Just as the name tag on Chan's spaceship provides essential details like the spaceship's name and model, the `info` section in AsyncAPI serves a similar purpose. It provides crucial information about the API, such as the title, version, and description. This metadata helps developers understand the basic details of the API they're working with.
+
+```yml
+asyncapi: '3.0.0'
+info:
+  title: Space Communication API
+  version: '1.0.0'
+  description: API for communication between Chan's spaceship and Capuccinova
+```
```
Comment (quetzalliwrites) [inline 100-level-AsyncAPI-Intro/step2.md]:
```suggestion
- **Info**

Just as the name tag on Chan's spaceship provides essential details like the spaceship's name and model, the `info` section in AsyncAPI serves a similar purpose. It provides crucial information about the API, such as the title, version, and description. The `info` metadata helps developers understand the basic API details they're working with.

```yml
asyncapi: '3.0.0'
info:
  title: Space Communication API
  version: '1.0.0'
  description: API for communication between Chan's spaceship and Capuccinova
```
```

File: 100-level-AsyncAPI-Intro/step2.md [hunk 4]
```diff
@@ -0,0 +1,244 @@
...
+
+Imagine that Chan's spaceship communicates with Eve on Capuccinova through various servers. In AsyncAPI, the `servers` section describes the servers our API will communicate with, including their hosts, protocols, and descriptions.
+
+```yml
+servers:
+  production:
+    host: space.mosquitto.org
+    protocol: mqtt
+    description: Production server for space communication
+```
```
Comment (quetzalliwrites) [inline 100-level-AsyncAPI-Intro/step2.md]:
```suggestion
- **Servers**

Imagine that Chan's spaceship communicates with Eve on Capuccinova through various servers. In AsyncAPI, the `servers` section describes the servers our API will communicate with, including their hosts, protocols, and descriptions.

```yml
servers:
  production:
    host: space.mosquitto.org
    protocol: mqtt
    description: Production server for space communication
```
```

File: 100-level-AsyncAPI-Intro/step2.md [hunk 5]
```diff
@@ -0,0 +1,244 @@
...
+
+- **Channels (Communication channels in the API):**
```
Comment (quetzalliwrites) [inline 100-level-AsyncAPI-Intro/step2.md]:
```suggestion
- **Channels**
```

File: 100-level-AsyncAPI-Intro/step2.md [hunk 6]
```diff
@@ -0,0 +1,244 @@
...
+            userId:
+              type: string
+              description: The ID of the user who sent the message
+            message:
+              type: string
+              description: The chat message
+            spacename:
+              type: string
+              description: The unique spacename identifier for the message
+```
```
Comment (quetzalliwrites) [inline 100-level-AsyncAPI-Intro/step2.md]:
```suggestion
The spaceship's radio stations, which correspond to specific topics like communicating with Capuccinova, are similar to the `channels` in AsyncAPI. These channels represent paths for sending specific types of messages. Each channel uses an address, which is a template for the URL of the server, and can have variables inside curly braces.

```yml
channels:
  roomidchat:
    address: chat/{roomId}
    messages:
      chatMessage:
        name: ChatMessage
        payload:
          type: object
          properties:
            userId:
              type: string
              description: The ID of the user who sent the message
            message:
              type: string
              description: The chat message
            spacename:
              type: string
              description: The unique spacename identifier for the message
```
```

File: 100-level-AsyncAPI-Intro/step2.md [hunk 7]
```diff
@@ -0,0 +1,244 @@
...
+
+- **Tags (Categorizing operations):**
+
+`Tags` in AsyncAPI are like the labels on the spaceship's control panel. They help categorize different operations, making it easier for developers to understand what each part of the API does. Each tag consists of a name and an optional description.
+
+```yml
+tags:
+  - name: chat
+    description: Operations related to chat
+```
```
Comment (quetzalliwrites) [inline 100-level-AsyncAPI-Intro/step2.md]:
```suggestion
- **Tags**

`Tags` in AsyncAPI are like the labels on the spaceship's control panel. They help categorize different operations, making it easier to understand what each part of the API does. Each tag consists of a name and an optional description.

```yml
tags:
  - name: chat
    description: Operations related to chat
```
```

File: 100-level-AsyncAPI-Intro/step2.md [hunk 8]
```diff
@@ -0,0 +1,244 @@
...
+    ChatMessagePayload:
+      type: object
+      properties:
+        userId:
+          type: string
+          description: The ID of the user who sent the message
+        message:
+          type: string
+          description: The chat message
+```
```
Comment (quetzalliwrites) [inline 100-level-AsyncAPI-Intro/step2.md]:
```suggestion
- **Components**

`Component` is like the toolbox in Chan's spaceship. It contains reusable parts like message definitions and schemas that can be used across the API. Each component is linked to a particular schema, which describes the structure of the message content.

```yml
components:
  messages:
    ChatMessage:
      name: ChatMessage
      title: Chat Message
      contentType: application/json
      payload:
        $ref: '#/components/schemas/ChatMessagePayload'
  schemas:
    ChatMessagePayload:
      type: object
      properties:
        userId:
          type: string
          description: The ID of the user who sent the message
        message:
          type: string
          description: The chat message
```
```

File: 100-level-AsyncAPI-Intro/step2.md [hunk 9]
```diff
@@ -0,0 +1,244 @@
...
+
+## Programming Operations, Messages, and Schemas
```
Comment (quetzalliwrites) [inline 100-level-AsyncAPI-Intro/step2.md]:
```suggestion
## Operations, messages, and schemas
```

File: 100-level-AsyncAPI-Intro/step2.md [hunk 10]
```diff
@@ -0,0 +1,244 @@
...
+`Operations` in AsyncAPI are like the instructions for the spaceship. They define what actions can be performed within each channel, such as sending or receiving messages. Each operation is associated with a message and a schema.
+
+```yml
+operations:
+  sendChatMessage:
+    action: 'send'
+    summary: Send a chat message to a specific room
+    channel:
+      $ref: '#/channels/roomidchat'
+```
```
Comment (quetzalliwrites) [inline 100-level-AsyncAPI-Intro/step2.md]:
```suggestion
- **Operations**

`Operations` in AsyncAPI are like the instructions for the spaceship. They define what actions can be performed within each channel, such as sending or receiving messages. Each operation is associated with a message and a schema.

```yml
operations:
  sendChatMessage:
    action: 'send'
    summary: Send a chat message to a specific room
    channel:
      $ref: '#/channels/roomidchat'
```
```

File: 100-level-AsyncAPI-Intro/step2.md [hunk 11]
```diff
@@ -0,0 +1,244 @@
...
+```yml
+components:
+  messages:
+    ChatMessage:
+      name: ChatMessage
+      title: Chat Message
+      contentType: application/json
+      payload:
+        $ref: '#/components/schemas/ChatMessagePayload'
+```
```
Comment (quetzalliwrites) [inline 100-level-AsyncAPI-Intro/step2.md]:
```suggestion
- **Messages**

`Messages` in AsyncAPI are the content exchanged during operations, just like the words Eve from Capuccinova and Chan from Brownieterra use to communicate. They describe the structure and content of the messages sent or received.


```yml
components:
  messages:
    ChatMessage:
      name: ChatMessage
      title: Chat Message
      contentType: application/json
      payload:
        $ref: '#/components/schemas/ChatMessagePayload'
```
```

File: 100-level-AsyncAPI-Intro/step2.md [hunk 12]
```diff
@@ -0,0 +1,244 @@
...
+    ChatMessagePayload:
+      type: object
+      properties:
+        userId:
+          type: string
+          description: The ID of the user who sent the message
+        message:
+          type: string
+          description: The chat message
+```
```
Comment (quetzalliwrites) [inline 100-level-AsyncAPI-Intro/step2.md]:
```suggestion
- **Schemas**

`Schemas` in AsyncAPI are like the blueprints for the spaceship. They define the structure of the message content, ensuring that everything fits together correctly. Each schema is associated with a component and describes the structure and content of the messages.

```yml
components:
  schemas:
    ChatMessagePayload:
      type: object
      properties:
        userId:
          type: string
          description: The ID of the user who sent the message
        message:
          type: string
          description: The chat message
```
```

File: 100-level-AsyncAPI-Intro/step2.md [hunk 13]
```diff
@@ -0,0 +1,244 @@
...
+
+---
+
+Certainly! Here's the updated hands-on exercise, guiding you through the entire process of creating and configuring Chan's AsyncAPI document for communication with Capuccinova and beyond.
+
```
Comment (quetzalliwrites) [inline 100-level-AsyncAPI-Intro/step2.md]: @mhmohona This line shows clear plagiarism from chatgpt.

File: 100-level-AsyncAPI-Intro/step2.md [hunk 14]
```diff
@@ -0,0 +1,244 @@
...
+### Chan's Galactic Coding Adventure: Crafting the Capuccinova Connection
+
+#### Prologue: A New Beginning in the Stars
+
+Chan embarks on a mission to establish a sophisticated communication system with Eve on Capuccinova. With your expertise in AsyncAPI, you'll assist him in creating a robust AsyncAPI document from scratch.
+
+---
```
Comment (quetzalliwrites) [inline 100-level-AsyncAPI-Intro/step2.md]:
```suggestion
Now it's time to help Chan establish a sophisticated communication system with Eve on Capuccinova. With your new expertise in AsyncAPI, you'll assist him in creating an AsyncAPI document from scratch.

```

File: 100-level-AsyncAPI-Intro/step2.md [hunk 15]
```diff
@@ -0,0 +1,244 @@
...
+
+#### Chapter 1: Initiating the Capuccinova Communication Link
```
Comment (quetzalliwrites) [inline 100-level-AsyncAPI-Intro/step2.md]:
```suggestion
#### Exercise 1: Create AsyncAPI document
```

File: 100-level-AsyncAPI-Intro/step2.md [hunk 16]
```diff
@@ -0,0 +1,244 @@
...
+
+**Your Mission: Creating the Base AsyncAPI Document**
```
Comment (quetzalliwrites) [inline 100-level-AsyncAPI-Intro/step2.md]:
```suggestion
```

File: 100-level-AsyncAPI-Intro/step2.md [hunk 17]
```diff
@@ -0,0 +1,244 @@
...
+
+1. **Start the Journey**: In the KillerCoda terminal, create a new file by typing `nano chan-to-capuccinova.yaml`.
```
Comment (quetzalliwrites) [inline 100-level-AsyncAPI-Intro/step2.md]:
```suggestion
1. In the KillerCoda terminal, create a new file by typing `nano chan-to-capuccinova.yaml`.
```

File: 100-level-AsyncAPI-Intro/step2.md [hunk 18]
```diff
@@ -0,0 +1,244 @@
...
+
+2. **Lay the Cosmic Foundations**: Enter the basic AsyncAPI structure:
```
Comment (quetzalliwrites) [inline 100-level-AsyncAPI-Intro/step2.md]:
```suggestion
2. Enter the following AsyncAPI document structure:
```

File: 100-level-AsyncAPI-Intro/step2.md [hunk 19]
```diff
@@ -0,0 +1,244 @@
...
+   operations:
+     sendChatMessage:
+       action: 'send'
+       summary: Send a chat message to a specific room
+       channel:
+         $ref: '#/channels/roomidchat'
+   ```
+3. **Seal the Galactic Document**: Save (`Ctrl + O`, `Enter`) and exit Nano (`Ctrl + X`).
+
+4. **Inform Chan**: Let him know that the foundation for his AsyncAPI document is set.
```
Comment (quetzalliwrites) [inline 100-level-AsyncAPI-Intro/step2.md]:
```suggestion
```

File: 100-level-AsyncAPI-Intro/step2.md [hunk 20]
```diff
@@ -0,0 +1,244 @@
...
+
+#### Chapter 2: Expanding the Network to Brownieterra
+
+**Your Mission: Adding a New Channel and Defining Operations for Brownieterra**
+
+After successfully laying the groundwork for communication with Capuccinova, Chan is now eager to extend his AsyncAPI document to include a channel for Brownieterra.
```
Comment (quetzalliwrites) [inline 100-level-AsyncAPI-Intro/step2.md]:
```suggestion
#### Exercise 2: Add new channel and define operations**

After successfully laying the groundwork for communication with Capuccinova, Chan is eager to extend his AsyncAPI document to include a channel for Brownieterra. 
```

File: 100-level-AsyncAPI-Intro/step2.md [hunk 21]
```diff
@@ -0,0 +1,244 @@
...
+     sendToBrownieterra:
+       action: send
+       summary: Send a message to Brownieterra
+       channel:
+         $ref: '#/channels/brownieterra-chat'
+   ```
+
+4. **Save and Exit**: After adding the new channel and defining its operations, save your changes (`Ctrl + O`, `Enter`) and exit Nano (`Ctrl + X`).
+
+5. **Inform Chan**: Let Chan know that the new channel for Brownieterra has been successfully added and configured in his AsyncAPI document.
```
Comment (quetzalliwrites) [inline 100-level-AsyncAPI-Intro/step2.md]:
```suggestion
1. Run the command `nano chan-to-capuccinova.yaml` to edit Chan's document.
2. Add a channel for Brownieterra, specifying the channel name, its address, and parameters.
```yaml
   brownieterra-chat:
     description: Channel for sending messages to Brownieterra
     address: brownieterra-chat/{spacename}
     parameters:
       spacename:
         description: The unique spacename identifier for Brownieterra communication
         schema:
           type: string
```
3. Under the `operations` section, specify how messages will be sent over the new Brownieterra channel.
  ```yaml
   operations:
     sendToBrownieterra:
       action: send
       summary: Send a message to Brownieterra
       channel:
         $ref: '#/channels/brownieterra-chat'
  ```
```

File: 100-level-AsyncAPI-Intro/step2.md [hunk 22]
```diff
@@ -0,0 +1,244 @@
...
+
+Chan wants to add more details to the message payloads for more personalized communication.
+
+1. **Access the AsyncAPI File**: Edit `chan-to-capuccinova.yaml` in Nano.
+
+2. **Refine the Message Payloads**: Update the payloads to include additional properties or modify existing ones to better suit the communication needs.
+
+3. **Consolidate the Changes**: Save your modifications (`Ctrl + O`, `Enter`) and exit (`Ctrl + X`).
+
+4. **Update Chan**: Let him know that the message formats are now more detailed and well structured for both Capuccinova and Brownieterra.
```
Comment (quetzalliwrites) [inline 100-level-AsyncAPI-Intro/step2.md]:
```suggestion
```

This is not an exercise as no steps were added. Removing.

**Review by smoya — CHANGES_REQUESTED** [review]
I noticed there are 3 PR's opened pointing to the exact same file `step2.md`. 

- https://github.com/asyncapi/learning-paths/pull/5
- https://github.com/asyncapi/learning-paths/pull/9
- https://github.com/asyncapi/learning-paths/pull/11

Please unify all of those into one single PR so we can review the file as a whole. 

cc @TRohit20 @mhmohona @alequetzalli

**Comment by quetzalliwrites** [comment]
> I noticed there are 3 PR's opened pointing to the exact same file `step2.md`.
> 
> * [docs: add content for module 3 #5](https://github.com/asyncapi/learning-paths/pull/5)
> * [docs: adding content for module 3 #9](https://github.com/asyncapi/learning-paths/pull/9)
> * [docs: adding request/reply content in module-3 #11](https://github.com/asyncapi/learning-paths/pull/11)
> 
> Please unify all of those into one single PR so we can review the file as a whole.

I hear you, let's follow your suggestion 😸 .  @TRohit20, can you please handle this one?

**Comment by TRohit20** [comment]
@alequetzalli Sure, I'd just have to close all the PRs to begin with and then merge the content into a single document which is anyhow the final output and raise a PR reflecting it.

Would you like me to proceed with that? @alequetzalli + I'm unsure if I have the repo right to close @mhmohona's PR

**Comment by mhmohona** [comment]
I have closed my PR, does it help @TRohit20?

**Pull request closed without merging**